### PR TITLE
imapc: Fix memory leak in disconnected state

### PR DESCRIPTION
**Pull request by zeil**

### CODE REVIEW

File: src/lib-imap-client/imapc-connection.c [hunk 1]
```diff
@@ -382,6 +382,10 @@ static void imapc_connection_set_state(struct imapc_connection *conn,
 
 		conn->selecting_box = NULL;
 		conn->selected_box = NULL;
+
+		if (conn->ips != NULL)
```
Comment (jeffpc) [inline src/lib-imap-client/imapc-connection.c]: IIRC, `i_free(NULL)` is a no-op so there is no need for this NULL check.

Comment (jeffpc) [inline src/lib-imap-client/imapc-connection.c]: Hrm, looking at the implementation of it, I may be wrong - it might try to memset the memory first.

Comment (sirainen) [inline src/lib-imap-client/imapc-connection.c]: i_free(NULL) is no-op, just like free(NULL) in ANSI-C.

Comment (sirainen) [inline src/lib-imap-client/imapc-connection.c]: ..and it's used all over the code everywhere, so better to be consistent here also and not have the if-check :)

**Comment by cmouse** [comment]
Merged outside github.

**Pull request closed without merging**